src/gui/guiFormSpecMenu.h

```diff
@@ -278,12 +280,15 @@ class GUIFormSpecMenu : public GUIModalMenu
 
 	std::unordered_map<std::string, std::vector<StyleSpec>> theme_by_type;
 	std::unordered_map<std::string, std::vector<StyleSpec>> theme_by_name;
+	std::unordered_map<std::string, std::vector<StyleSpec>> theme_by_group;
```

Comment:
Is it necessary for this to be a `std::vector<StyleSpec>`?  It seems that it could be a plain `StyleSpec`, and the new values could just be ORed in, since they're getting ORed eventually already in `getStyleForElement`.  Or is there another reason for this?

Reply:
If I did this, it was probably because individual styles in these lists can be associated with different sets of states. I think you could probably replace the vector with an array matching the number of states but not a single style object.

Reply:
Ah, now I see.  I misunderstood the code.  Yes, a `std::vector` would probably be better here because not all the states might exist for a single selector, and for non-state elements, it will always be one.